library/core/benches/iter.rs

```diff
@@ -367,3 +367,54 @@ fn bench_partial_cmp(b: &mut Bencher) {
 fn bench_lt(b: &mut Bencher) {
     b.iter(|| (0..100000).map(black_box).lt((0..100000).map(black_box)))
 }
+
+#[bench]
+fn bench_array_map(b: &mut Bencher) {
+    b.iter(|| {
+        let mut acc = 0;
+        let iter = (0i64..10000).map(black_box).map(|_| black_box([0i64; 100]));
+        for_each_fold(iter, |x| acc += x.iter().sum::<i64>());
```

Comment:
`x.iter()` on an array gives a slice iter, have you tried whether an `array::IntoIter` performs better or worse?

Reply:
The closure here is the same as the closures in the other benchmarks. With the exception of the slice `array_chunks` benchmark (which I have just removed), the type of `x` in each closure is an array, so they should be comparable.

Reply:
Ok, I was asking in case it made them faster and the smaller baseline resulting in a wider gap.